comp/core/configsync/configsyncimpl/module.go

```diff
@@ -88,6 +100,19 @@ func newConfigSync(deps dependencies, agentIPCPort int, configRefreshIntervalSec
 		ctx:       ctx,
 	}
 
+	if deps.SyncParams.SyncOnInit {
+		if deps.SyncParams.SyncDelay != 0 {
```

Comment:
Do we need a delay here ? Would it not be better for the sync to retry indefinitely until we can contact the core agent ?

Reply:
We need the delay for the same reason we can't use lifecycle hooks. We need the sync to happen during the instantiation so that when the `config` is injected it has already been synced. The delay helps us address the race-condition with the core-agent. Again, something we can hopefully clean up once we can propagate config changes in the `otel-agent`.